### PR TITLE
Advertise mdakits

### LINKED ISSUE #4296: Advertise MDAKits
It would be helpful to have a link to the MDAKits registry in the official MDAnalysis docs.

Requires https://github.com/MDAnalysis/mdanalysis-sphinx-theme/issues/67 to do so in the navbar easily.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #4296

Changes made in this Pull Request:
 - Updates to 1.3.0, adds sidebar links

Waiting on 1.3.0 to get onto conda-forge.

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4297.org.readthedocs.build/en/4297/

### CODE REVIEW

**Comment by lilyminium** [comment]
Screenshot 2023-09-22 at 4 55 07 pm


Docs wind up looking like this with the new header (it looked kind of weird without the "Documentation" header in orange)

**Comment by IAlibay** [comment]
Whilst I'm here - not this PR but, is the first link really meant to say "Overview over MDAnalysis"?

**Comment by lilyminium** [comment]
> is the first link really meant to say "Overview over MDAnalysis"?

It's pulling the title from https://github.com/MDAnalysis/mdanalysis/blob/develop/package/doc/sphinx/source/documentation_pages/overview.rst

**Comment by IAlibay** [comment]
> > is the first link really meant to say "Overview over MDAnalysis"?
> 
> It's pulling the title from https://github.com/MDAnalysis/mdanalysis/blob/develop/package/doc/sphinx/source/documentation_pages/overview.rst

I guess I should clarify my question - I assume it's historical but it doesn't read amazingly well. We don't need to tackle this in this PR, but I would be keen to know if it's just me thinking it's odd or not.

File: .github/workflows/gh-ci.yaml [hunk 1]
```diff
@@ -155,7 +155,7 @@ jobs:
       with:
         micromamba: true
         full-deps: true
-        extra-pip-deps: "docutils sphinx<7 sphinx-sitemap mdanalysis-sphinx-theme>=1.0.1 sphinxcontrib-bibtex pybtex pybtex-docutils"
+        extra-pip-deps: "docutils sphinx<7 sphinx-sitemap mdanalysis-sphinx-theme>=1.3.0 sphinxcontrib-bibtex pybtex pybtex-docutils"
```
Comment (IAlibay) [inline .github/workflows/gh-ci.yaml]: Note; if these are conda installable now we can just switch this to `extra-conda-deps` to speed up the build.

**Review by IAlibay — APPROVED** [review]
lgtm otherwise

**Pull request merged**